infra/modules/azure_role_assignments/variables.tf

```diff
@@ -0,0 +1,87 @@
+variable "principal_id" {
+  description = "The ID of the principal to which assign roles. It can be a managed identity."
+  type        = string
+}
+
+variable "cosmos" {
+  description = "A list of CosmosDB role assignments"
+  type = list(object({
+    account_name        = string
+    resource_group_name = string
+    role                = string
+    database            = optional(string, "*")
+    collection          = optional(string, "*")
```

Comment:
Should we use a `list(string)` for both databases and collections?

Reply:
Good point! Initially, I actually started with a list of strings. However, the code became a little more intricate, so I thought that, if the caller needs to add permissions for different databases and collections, would do something like this:

```
module "roles" {
  source       = "../../"
  principal_id = module.function.function_app.function_app.principal_id

  cosmos = [
    {
      account_name        = "io-p-cosmos-example-cms"
      resource_group_name = "io-p-example-cms-rg"
      role                = "writer"
      database            = "a"
      collection          = "x"
    },
    {
      account_name        = "io-p-cosmos-example-cms"
      resource_group_name = "io-p-example-cms-rg"
      role                = "writer"
      database            = "a"
      collection          = "y"
    }
  ]
}
```

I see that it can become large quickly, so the alternative can be to make the collection parameter as a list(string). I wouldn't make the database parameter a list(string) because we would lose the database => collection relationship.
In this case, it would look like this which looks better indeed:

```
module "roles" {
  source       = "../../"
  principal_id = module.function.function_app.function_app.principal_id

  cosmos = [
    {
      account_name        = "io-p-cosmos-example-cms"
      resource_group_name = "io-p-example-cms-rg"
      role                = "writer"
      database            = "a"
      collection          = ["x", "y"]
    }
  ]
}
```